test/battle/move_effect/protect.c

```diff
@@ -0,0 +1,369 @@
+                    }
+                }
+            }
+        }
+    } THEN {
+        if (usedMove == MOVE_TACKLE) {
+            EXPECT_EQ(player->statStages[statId], DEFAULT_STAT_STAGE - lowersBy);
+        }
+    }
+}
+
+// Also checks if it can faint the mon.
+SINGLE_BATTLE_TEST("Spiky Shield does 1/8 dmg of max hp for moves making contact")
+{
+    u16 usedMove = MOVE_NONE;
+    u16 hp = 400, maxHp = 400;
+
+    PARAMETRIZE {usedMove = MOVE_TACKLE; hp = 1;}
+    PARAMETRIZE {usedMove = MOVE_TACKLE;}
+    PARAMETRIZE {usedMove = MOVE_LEER;}
+    PARAMETRIZE {usedMove = MOVE_WATER_GUN;}
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET) {HP(hp); MaxHP(maxHp); }
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        if (hp == 1) {
+            TURN { MOVE(opponent, MOVE_SPIKY_SHIELD); MOVE(player, usedMove); SEND_OUT(player, 1); }
+        } else {
+            TURN { MOVE(opponent, MOVE_SPIKY_SHIELD); MOVE(player, usedMove); }
+        }
+        TURN {}
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SPIKY_SHIELD, opponent);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT ANIMATION(ANIM_TYPE_MOVE, usedMove, player);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT HP_BAR(opponent);
+        if (usedMove == MOVE_TACKLE) {
+            HP_BAR(player, maxHp / 8);
+            if (hp == 1) {
+                MESSAGE("Wobbuffet fainted!");
+                MESSAGE("Go! Wobbuffet!");
+            }
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Baneful Bunker poisons pokemon for moves making contact")
+{
+    u16 usedMove = MOVE_NONE;
+
+    PARAMETRIZE {usedMove = MOVE_TACKLE; }
+    PARAMETRIZE {usedMove = MOVE_LEER; }
+    PARAMETRIZE {usedMove = MOVE_WATER_GUN; }
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_BANEFUL_BUNKER); MOVE(player, usedMove); }
+        TURN {}
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_BANEFUL_BUNKER, opponent);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT ANIMATION(ANIM_TYPE_MOVE, usedMove, player);
+        MESSAGE("Foe Wobbuffet protected itself!");
+        NOT HP_BAR(opponent);
+        if (usedMove == MOVE_TACKLE) {
+            STATUS_ICON(player, STATUS1_POISON);
+        } else {
+            NOT STATUS_ICON(player, STATUS1_POISON);
+        }
+    }
+}
+
+SINGLE_BATTLE_TEST("Recoil is not applied if battler was protected") // #3319
+{
+    u16 move = MOVE_NONE;
+
+    PARAMETRIZE {move = MOVE_PROTECT; }
+    PARAMETRIZE {move = MOVE_DETECT; }
+    PARAMETRIZE {move = MOVE_KINGS_SHIELD; }
+    PARAMETRIZE {move = MOVE_BANEFUL_BUNKER; }
+    PARAMETRIZE {move = MOVE_SILK_TRAP; }
+    PARAMETRIZE {move = MOVE_OBSTRUCT; }
+    PARAMETRIZE {move = MOVE_SPIKY_SHIELD; }
+
+    GIVEN {
+        ASSUME(gBattleMoves[MOVE_DOUBLE_EDGE].effect == EFFECT_RECOIL_33);
```

Comment:
Let's add the other `EFFECT_RECOIL_x`s here

Reply:
Done